### PR TITLE
Revert changes that allowed differentiation to look across cells in REPL

### PR DESCRIPTION
**Pull request by bgogul**
This is one of the changes required to fix https://bugs.swift.org/browse/TF-9

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 1]
```diff
@@ -1796,15 +1790,10 @@ emitAssociatedFunctionReference(ADContext &context, SILBuilder &builder,
         context.lookUpMinimalDifferentiationTask(originalFn, desiredIndices);
     if (!task) {
       if (originalFn->isExternalDeclaration()) {
-        // For LLDB REPL, we should attempt to load the function as
-        // this may be defined in a different cell.
-        if (isInLLDBREPL(*original->getModule()))
-          original->getModule()->loadFunction(originalFn);
-        // If we still don't have the definition, generate an error message.
-        if (originalFn->isExternalDeclaration()) {
+        if (!originalFn->isDefinition()) {
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: Note: the [old condition](https://github.com/apple/swift/commit/01339e7832297d00d7124d1e1ea0cf89aa8dde99#diff-4ede198777b59748c377cdd17c444e83L1653) here seems only to check for `if (originalFn->isExternalDeclaration())`.

**Review by dan-zheng — APPROVED** [review]
Yay!

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 2]
```diff
@@ -1796,15 +1790,10 @@ emitAssociatedFunctionReference(ADContext &context, SILBuilder &builder,
...
           context.emitNondifferentiabilityError(
-              original, parentTask,
-              diag::autodiff_external_nondifferentiable_function);
+            original, parentTask,
```
Comment (rxwei) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: Indent by 4.

**Review by rxwei — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/Differentiation.cpp (hunk 1 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/Differentiation.cpp]: Aren't they equivalent? In any case,  I will change this to the old condition to be consistent with the rest of the codebase.

File: lib/SILOptimizer/Mandatory/Differentiation.cpp (hunk 2 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/Differentiation.cpp]: Done

File: lib/SILOptimizer/Mandatory/Differentiation.cpp (hunk 1 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/Differentiation.cpp]: Done.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
@swift-ci please clean test tensorflow

**Pull request merged**